Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/FreightAnimations.cs

```diff
@@ -302,6 +309,21 @@ public void Load(string loadFilePath, LoadPosition loadPosition)
                     container.LoadFromContainerFile(loadFilePath, Wagon.Simulator.BasePath +@"\trains\trainset\");
                     ContainerManager.LoadedContainers.Add(loadFilePath, container);
                 }
+                switch (loadState)
+                {
+                    case LoadState.Empty:
+                        container.MassKG = container.EmptyMassKG;
+                        break;
+                    case LoadState.Loaded:
+                        container.MassKG = container.MaxMassWhenLoadedKG;
+                        break;
+                    case LoadState.Random:
+                        var loadPercent = Simulator.Random.Next(101);
+                        if (loadPercent < 30) container.MassKG = container.EmptyMassKG;
+                        else container.MassKG = container.MaxMassWhenLoadedKG * loadPercent / 100f;
+                        break;
+                }
+
```

Comment:
Same comment as before regarding the linear map. Please consider also eliminating the code duplication.

Reply:
Code duplication removed.